### ISSUE TITLE
different shape test

### ISSUE THREAD

**Issue by skdingding** [MShahzebKhan/Spatial-and-Angular-Resolution-Enhancement-of-Light-Fields-Using-Convolutional-Neural-Networks#1]
hi,I want to test my own lf image,but my image shape is not same as yours.mine is 8*8*512*512,i fail to make it,can you help me? it may be some error in fc1.

I0121 21:24:27.120545 19296 net.cpp:255] Network initialization done.
F0121 21:24:27.145087 19296 net.cpp:757] Cannot copy param 0 weights from layer 'fc1'; shape mismatch.  Source param shape is 196 800 (156800); target param shape is 64 128 (8192). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
*** Check failure stack trace: ***

**Comment by MShahzebKhan** [comment]
> 
> 
> hi,I want to test my own lf image,but my image shape is not same as yours.mine is 8_8_512*512,i fail to make it,can you help me? it may be some error in fc1.
> 
> I0121 21:24:27.120545 19296 net.cpp:255] Network initialization done.
> F0121 21:24:27.145087 19296 net.cpp:757] Cannot copy param 0 weights from layer 'fc1'; shape mismatch. Source param shape is 196 800 (156800); target param shape is 64 128 (8192). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
> *** Check failure stack trace: ***

Yes because they are trained for the Lytro Illum light field data and due to the fully connected layer you can not use it directly for any other dataset, until and unless it has the same angular resolution (14x4).

To use it for any other dataset, you have to retrain the model.

**Comment by yyyysl** [comment]
Hi, I tried to decode the light field image (.lfr format) using LFtoolbox0.4 and lfr2png, and the raw image generated is different from the input light field image you gave. Below each microlens is 14 * 14 (angular resolution), but in a raw image each microlens is staggered, with two adjacent rows of microlenses not in the same column. I download the test input light field images from the link, but I do not know how to generate  light field images(like yours) in.lfr format. can you help me?
